pkg/domain/stateDiff.go

```diff
@@ -1,173 +1,20 @@
 package domain
 
-import (
-	"fmt"
-	"github.com/cloudogu/cesapp-lib/core"
-	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
-	"golang.org/x/exp/maps"
-)
-
 // StateDiff represents the diff between the defined state in the effective blueprint and the actual state in the ecosystem.
 // If there is a state in the ecosystem, which is not represented in the effective blueprint, then the expected state is the actual state.
 type StateDiff struct {
-	DoguDiffs DoguDiffs
-}
-
-// DoguDiffs contains the Diff for all expected Dogus to the current ecosystem.DoguInstallations.
-type DoguDiffs []DoguDiff
-
-// Statistics aggregates various figures about the required actions of the DoguDiffs.
-func (dd DoguDiffs) Statistics() (toInstall int, toUpgrade int, toUninstall int, other int) {
-	for _, doguDiff := range dd {
-		switch doguDiff.NeededAction {
-		case ActionInstall:
-			toInstall += 1
-		case ActionUpgrade:
-			toUpgrade += 1
-		case ActionUninstall:
-			toUninstall += 1
-		default:
-			other += 1
-		}
-	}
-	return
-}
-
-// DoguDiff represents the Diff for a single expected Dogu to the current ecosystem.DoguInstallation.
-type DoguDiff struct {
-	DoguName     string
-	Actual       DoguDiffState
-	Expected     DoguDiffState
-	NeededAction Action
-}
-
-// DoguDiffState contains all fields to make a diff for dogus in repect to another DoguDiffState.
-type DoguDiffState struct {
-	Namespace         string
-	Version           core.Version
-	InstallationState TargetState
+	DoguDiffs      DoguDiffs
+	ComponentDiffs ComponentDiffs
 }
 
 // Action represents a needed Action for a dogu to reach the expected state.
 type Action string
 
 const (
-	ActionNone            = "none"
-	ActionInstall         = "install"
-	ActionUninstall       = "uninstall"
-	ActionUpgrade         = "upgrade"
-	ActionDowngrade       = "downgrade"
-	ActionSwitchNamespace = "namespace switch"
+	ActionNone                = "none"
+	ActionInstall             = "install"
+	ActionUninstall           = "uninstall"
+	ActionUpgrade             = "upgrade"
+	ActionDowngrade           = "downgrade"
+	ActionSwitchDoguNamespace = "dogu namespace switch"
```

Comment:
In my eyes the whole dogu namespace switch feature boggles my mind, big case! While I AM able to project that this might be a thing, I won't support this out of the box because it must mean the following:

1. Develop a component, push it into harbor project A
2. develop the same component, push it into a differerent harbor project B
3. rollout both versions
4. ?????
5. profit and switcheroo all around because thats a perfectly normal thing to do?